astropy/units/core.py

```diff
@@ -687,27 +697,31 @@ def aliases(self):
         )
 
     @property
-    def scale(self):
+    def scale(self) -> UnitScale:
         """
         Return the scale of the unit.
         """
         return 1.0
 
     @property
-    def bases(self):
+    def bases(self) -> list[UnitBase]:
         """
         Return the bases of the unit.
         """
         return [self]
 
     @property
-    def powers(self):
+    def powers(self) -> list[UnitPower]:
         """
         Return the powers of the unit.
         """
         return [1]
 
-    def to_string(self, format=unit_format.Generic, **kwargs):
+    def to_string(
+        self,
+        format: type[unit_format.Base] | str | None = unit_format.Generic,
```

Comment:
There are two good reasons for allowing `None` here.

1. There is code in `astropy` that can call `UnitBase.to_string()` with `format=None`, e.g. https://github.com/astropy/astropy/blob/6130037fc72111ec24c31421884b7859f2b02745/astropy/coordinates/angles/core.py#L356

2. Calling `UnitBase.to_string()` with `format=None` leads to fastest execution at runtime because this function calls `astropy.units.format.get_format()` and the first thing that function does is it checks https://github.com/astropy/astropy/blob/6130037fc72111ec24c31421884b7859f2b02745/astropy/units/format/__init__.py#L90 

Both of these points I brought up could be changed, but the annotation correctly documents the current situation.

Reply:
There are many dunder methods in this file with docstrings, even though dunder methods don't really need them because what they do is defined well enough by the [Python data model](https://docs.python.org/3/reference/datamodel.html) (`__new__()`, `__init__()` and `__call__()` are the three exceptions to this).

Additionally, there are many properties with docstrings like https://github.com/astropy/astropy/blob/6130037fc72111ec24c31421884b7859f2b02745/astropy/units/core.py#L680-L684
when it would be much better to have a docstring like
```python
     @property 
     def aliases(self): 
         """The alias (long) names for this unit.""" 
```

In short, there are many changes to be made to the docstrings here, not just the change you are suggesting.

My preference would be to limit `typing` pull requests only to annotations and to address whatever the annotations reveal in follow-up pull requests, but if you or @nstarman think differently then let me know and I can push more commits here.

Reply:
> My preference would be to limit `typing` pull requests only to annotations and to address whatever the annotations reveal in follow-up pull requests, but if you or @nstarman think differently then let me know and I can push more commits here.

As long as we don't forget, I'm happy either way! (And at my age, forgetting is likely enough that I typically try to fix things as I go...)

As said, though, if @nstarman is happy, I think we should just merge this rather than run CI another time just for a really minor docstring fix.

Reply:
Agreed!